### DISCUSSION TITLE
Load collection very slow

### DISCUSSION THREAD

**Discussion by heqins** [milvus-io/milvus#23222] [Q&A and General discussion]
I have a collection with index IVF_FLAT, vector dimension of 512, and using Milvus version 2.2.3.

I set up standalone server with Docker on both a Linux and Windows device. When I load the collection on my windows device, it took about 10 seconds, but when I load collection on Linux, it took 194 seconds. I don't understand why loading on the Linux device took so much longer.

Configuration:
  local: 6 cores CPU,16GB RAM, HDD, 3.8k+ data
  linux: 32 cores CPU, 64GB RAM, SSD, 7k+ data

**Comment by heqins** [comment]
correction: milvus version is 2.2.0

**Reply by yhmo** [reply]
Load speed mainly depends on the bandwidth between MinIO/S3 and Milvus. It just simply read data from MinIO/S3 into memory.

**Reply by heqins** [reply]
Thank you for the reply. Upon investigation, we discovered that the deployment was originally performed on a mechanical hard drive. However, upon upgrading the machine to an SSD and migrating the data, we noticed that the loading speed did not improve significantly and still takes minutes to load. Interestingly, we observed that loading speed is much faster, taking only a few seconds, when inserting new vector data into a new collection. I would like to know why this is happening and whether it is possible to fix it.

**Reply by yhmo** [reply]
"3.8k+ data" ------ Only 3800 rows in the collection? It should be very fast to load.

I just use v2.2.0 standalone test on my HDD disk. Create a collection and insert 10000 rows vector(512 dim), it only cost 1 ~ 2 seconds to load.
Steps:
1. download the yaml
`wget https://github.com/milvus-io/milvus/releases/download/v2.2.0/milvus-standalone-docker-compose.yml -O docker-compose.yml`

2. start milvus standalone
`docker-compose up -d`

3. use the following script to create a collection, insert 10000 rows, and load the collection:
```
import random
import time

from pymilvus import (
    connections,
    FieldSchema, CollectionSchema, DataType,
    Collection,
    utility,
)

_HOST = '127.0.0.1'
_PORT = '19530'

if __name__ == '__main__':
    connections.connect(host=_HOST, port=_PORT)

    dim = 512
    collection_name = "test"
    if utility.has_collection(collection_name):
        utility.drop_collection(collection_name)

    field1 = FieldSchema(name="id", dtype=DataType.INT64, is_primary=True, auto_id=False)
    field2 = FieldSchema(name="embedding", dtype=DataType.FLOAT_VECTOR, dim=dim)
    schema = CollectionSchema(fields=[field1, field2])
    collection = Collection(name=collection_name, schema=schema)
    print("\ncollection created:", collection_name)

    index_param = {
        "index_type": "IVF_FLAT",
        "params": {"nlist": 256},
        "metric_type": "L2"}
    collection.create_index("embedding", index_param)

    num = 10000
    data = [
        [i for i in range(num)],
        [[random.random() for _ in range(dim)] for _ in range(num)],
    ]
    collection.insert(data)
    collection.flush()
    print("Insert", num, "vectors")
    print("Collection row count:", collection.num_entities)

    start = time.time()
    collection.load()
    end = time.time()
    print("Load collection, time cost:", (end-start)*1000, "ms")
```
The script output shows the time cost of loading is 1 ~ 2 seconds:
```
collection created: test
Insert 10000 vectors
Collection row count: 10000
Load collection, time cost: 1580.3608894348145 ms

Process finished with exit code 0
```

**Reply by yhmo** [reply]
You can run this script on your machine to compare your client code.
